pulser-pasqal/pulser_pasqal/pasqal_cloud.py

```diff
@@ -183,37 +183,44 @@ def _fetch_result(
         self, submission_id: str, job_ids: list[str] | None
     ) -> tuple[Result, ...]:
         # For now, the results are always sampled results
+        full_results = self._query_result(submission_id)
+
+        if job_ids is None:
+            job_ids = list(full_results.keys())
+
+        results = []
+        for id in job_ids:
+            assert full_results[id] is not None, "Failed to fetch the results."
```

Comment:
Let's turn this into a proper `raise` statement, informing the user they can still get the result of successful jobs via the new method

Reply:
Let's also make it conditional on the status?

Reply:
Should be good

Reply:
Yup!